### ISSUE TITLE
Adding Initial Support for Benchmark Perf CI

### ISSUE THREAD

**Issue by vcanicTT** [tenstorrent/tt-forge-fe#333]
We should have benchmark folder where we add all benchmarking models that work on real-world datasets. For those models we want to track end-to-end performance and evaluation score. Also, we want to add a possibility to track those models through the CI. Both performance and evaluation score are passing criteria.

fyi @tt-mpantic @odjuricicTT @nobradovictt @nvukobratTT

**Comment by nvukobratTT** [comment]
@vcanicTT before making any significant change, can you create a doc describing in more detail how this will work? 

One general note from my side on this topic to think about. Benchmarking shouldn't be a separate set of modes, but rather tooling around them to additionally configure and track relevant information (acc, perf, etc).

**Comment by vcanicTT** [comment]
I don't think we need a document, it will work in a similar way it worked in PyBuda. I will make it as a separate folder, because later if we want we can easily push it as a separate repo.

**Comment by nvukobratTT** [comment]
> I don't think we need a document, it will work in a similar way it worked in PyBuda. I will make it as a separate folder, because later if we want we can easily push it as a separate repo.

There were quite a few issues with how those worked with PyBuda, mostly related to how we maintained these models. 

If this will become a new repo from the start, and the perf team will be the owner of maintaining and patching it, then I don't have any objections to not creating a brief design doc and discussing it. However, if we plan to track these from the FFE repo, it'll be good to hear other opinions as well how these should be structured and maintained over time. Once again, speaking from limitations we faced with structuring these things on Buda compiler.
